crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -238,43 +269,55 @@ impl<'a, 'b> EntityCommands<'a, 'b> {
     /// }
     /// # example_system.system();
     /// ```
-    pub fn insert(&mut self, component: impl Component) -> &mut Self {
-        self.commands.add(Insert {
-            entity: self.entity,
-            component,
-        });
-        self
+    pub fn insert<T: Component>(
+        &mut self,
+        component: T,
+    ) -> FallibleCommandConfig<'_, Insert<T>, Self> {
+        FallibleCommandConfig::new(
+            Insert {
+                entity: self.entity,
+                component,
+            },
+            self,
+        )
     }
 
     /// See [`EntityMut::remove_bundle`](crate::world::EntityMut::remove_bundle).
-    pub fn remove_bundle<T>(&mut self) -> &mut Self
+    pub fn remove_bundle<T>(&mut self) -> FallibleCommandConfig<'_, RemoveBundle<T>, Self>
```

Comment:
With signatures like this, can we still get back to `&mut self` trivial for things like calling `.id()`?

Reply:
Yep.
`FallibleCommandConfig` implements `DerefMut<Target = T>` where T is the enclosing command type (so either `EntityCommand` or `Command`).

The following is valid syntax:
```rust
fn system(mut commands: Commands) {
    let id = commands
        .entity(e)
        .remove_bundle::<MyBundle>()
        .id();
        .despawn();
}
```